### PR TITLE
feat: add StructuredOutputScorer for validating JSON outputs

### PR DESCRIPTION
**Pull request by dcramer**
## Summary
- Adds a new `StructuredOutputScorer` for evaluating structured JSON outputs from language models
- Supports both strict and fuzzy matching modes for flexible validation
- Includes comprehensive test coverage with 483 lines of tests

## Changes
- **New scorer implementation**: `src/scorers/structuredOutputScorer.ts` (362 lines)
  - Validates JSON structure and field values
  - Supports strict equality checking and fuzzy matching
  - Handles nested objects, arrays, and complex data structures
  - Configurable matching modes and validation options
  
- **Comprehensive test suite**: `src/scorers/structuredOutputScorer.test.ts` (483 lines)
  - Tests strict and fuzzy matching modes
  - Covers edge cases like null values, empty objects, and arrays
  - Tests partial matching and extra field handling
  - Validates custom matcher functionality

- **Integration updates**: 
  - Exported from `src/scorers/index.ts`
  - Added to main exports in `src/index.ts`

## Features
- **Matching modes**:
  - `strict`: Exact equality required (default)
  - `fuzzy`: Case-insensitive strings, numeric tolerance, regex patterns, subset matching
  - Custom function support for specialized validation logic

- **Configuration options**:
  - `requireAll`: Whether all expected fields must be present
  - `allowExtras`: Whether to allow additional fields beyond expected
  - `debug`: Enable detailed logging for troubleshooting

## Type of change
- ✨ New feature (non-breaking change which adds functionality)

## Testing
- [x] Comprehensive unit tests added (100% coverage of new code)
- [x] All existing tests pass
- [x] TypeScript type checking passes
- [x] Linting passes

## Related issues
- N/A

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,28 +167,30 @@ import { Factuality, ClosedQA } from "autoevals";
...
-  }
-
-  const { object } = await generateObject({
-    model,
-    prompt: `
+import { generateObject } from "ai";
+import { openai } from "@ai-sdk/openai";
+import { z } from "zod";
+
+const Factuality =
```
Comment (dcramer) [inline README.md:184]: not sure what is reformatting this into terrible js, but problem for future me

File: src/index.ts [hunk 2]
```diff
@@ -8,27 +8,7 @@ import "vitest";
 export type ToolCall = {
   // Core fields (required for basic usage)
   name: string;
-  arguments: Record<string, any>;
-
-  // Result and timing
```
Comment (dcramer) [inline src/index.ts:13]: this shouldve never been in here tbqh - LLM hallucinated adding it to our types
